### ISSUE TITLE
Can't make public method protected

### ISSUE THREAD

**Issue by gerob** [cartalyst/sentinel#80]
Hey guys, first off thanks for open sourcing this! I don't think it's actually breaking anything but my IDE warns me that the PermissibleTrait->createPermissions() can't be set to `protected` as the abstract method it is overriding is set to public.

```
/**
     * Creates a permissions object.
     *
     * @return \Cartalyst\Sentinel\Permissions\PermissionsInterface
     */
    protected function createPermissions()
    {
        $userPermissions = $this->permissions;

        $rolePermissions = [];

        foreach ($this->roles as $role) {
            $rolePermissions[] = $role->permissions;
        }

        return new static::$permissionsClass($userPermissions, $rolePermissions);
    }
```

And the overridden method.

```
    /**
     * Creates the permissions object.
     *
     * @return \Cartalyst\Sentinel\Permissions\PermissionsInterface
     */
    abstract public function createPermissions();
```

**Comment by brunogaspar** [comment]
Hey @gerob, it should be fine, but if you want to tackle a fix, go ahead and submit a PR to the `1.0` branch.

**Comment by gerob** [comment]
@brunogaspar Thanks, I submitted a PR. https://github.com/cartalyst/sentinel/pull/81
